shotover-proxy/tests/runner/hotreload_int_tests.rs

```diff
@@ -96,19 +107,115 @@ async fn test_hot_reload_with_old_instance_shutdown() {
         .start()
         .await;
 
-    // Wait for the old shotover to shut down
-    shotover_old.consume_remaining_events(&[]).await;
-
-    // Verify that new shotover is still running and can handle connections
+    // Verify that new shotover is running and can handle new connections
     let client_new = Client::open("valkey://127.0.0.1:6380").unwrap();
     let mut con_new = client_new.get_connection().unwrap();
 
     // Verify data persistence
-    let value: String = con_new.get("test_key").unwrap();
-    assert_eq!(value, "test_value");
+    let value: String = con_new.get("key_0").unwrap();
+    assert_eq!(value, "value_0");
+
+    // Track connection failures over multiple drain cycles
+    // The gradual shutdown drains 10% of initial connections (rounded up to at least 1)
+    // For 13 connections: 10% = 1.3 -> rounds to 2 connections per cycle
+    // Each drain cycle waits 10 seconds between drains
+    //
+    // Note: Low connection counts give non-ideal cycle counts due to rounding.
+    // For example, 13 connections requires 7 cycles to fully drain.
+    // With realistic connection counts (100+), a 10% drain rate consistently takes ~10 cycles.
+    let total_connections = connections.len();
+    let expected_drain_per_cycle =
+        std::cmp::max(1, (total_connections as f64 * 0.1).ceil() as usize);
+
+    // Calculate how many drain cycles we need to verify
+    // We'll verify enough cycles to drain at least 50% of connections
+    let num_cycles_to_verify = std::cmp::min(
+        5,
+        (total_connections as f64 / expected_drain_per_cycle as f64 / 2.0).ceil() as usize,
+    );
+
+    // Verify each drain cycle
+    for cycle in 1..=num_cycles_to_verify {
+        tokio::time::sleep(tokio::time::Duration::from_secs(10)).await;
+
+        let mut connections_failed = 0;
+        for (i, con) in connections.iter_mut().enumerate() {
+            if con.get::<_, String>(format!("key_{}", i)).is_err() {
+                connections_failed += 1;
+            }
+        }
+
+        let expected_drained = expected_drain_per_cycle * cycle;
+        assert!(
+            connections_failed >= expected_drained,
```

Comment:
My thinking is that since we have such long cycle times (10s), we should be able to assert on an exact number?
Can you give this a go and report back what happens?